src/test/operators/table_scan_test.cpp

```diff
@@ -301,6 +306,24 @@ TEST_P(OperatorsTableScanTest, SingleScanWithSortedSegmentEquals) {
   EXPECT_TABLE_EQ_UNORDERED(scan->get_output(), expected_result);
 }
 
+TEST_P(OperatorsTableScanTest, SingleScanWithSortedSegmentEqualsAllElementsEqualNull) {
```

Comment:
I believe the whole sorted scan is undertested. Even now, we do not test, e.g., LessThan. We probably should adapt the existing tests to cover sorted segments. ScanOnCompressedSegmentsValueGreaterThanMaxDictionaryValue looks like it would have caught this if we had a version with sorted segments.

Your decision if you want to add this in the PR or open an issue.

Reply:
I totally agree. As @Bouncner said, since his group (including @d-justen) is working on the SortedSegmentScan, they could take care of that.

Reply:
Is this tracked somewhere?

Reply:
No, I will open an issue if we decide to do it like that.